src/build/env.py

```diff
@@ -38,6 +69,24 @@ def _restore_env(self):  # type: () -> None
     def _get_env_path(self, path):  # type: (str) -> Optional[str]
         return sysconfig.get_path(path, vars=self._env_vars)
 
+    def _symlink_relative(self, path):  # type: (Optional[str]) -> None
+        if not path:  # pragma: no cover
+            return
+
+        prefix = sysconfig.get_config_var('prefix')
+        if prefix and path and path.startswith(prefix):
+            new_path = os.path.join(self._path, path[len(prefix + os.pathsep):])
+            if not os.path.exists(new_path):
+                try:
+                    os.makedirs(os.path.dirname(new_path))
+                except FileExistsError:
+                    pass
+                if fs_supports_symlink():
+                    os.symlink(path, new_path)
+                else:
+                    import shutil
+                    shutil.copytree(path, new_path)
+
```

Comment:
In an ideal world would be nice to separate the test change and this functionality change here 🤔  As in separate PRs but 🤷 👍

Reply:
I'll do it in a separate PR.

Reply:
I'm ok for this to go in together now, but for future reference.

Reply:
Yes, I would also like for the test to go with the code, but this PR is huge :grin:. And the code was copied from virtualenv and only slightly modified, so I am dencently confident with the implementation, but I still want a test.